crates/pathfinder/src/rpc/types.rs

```diff
@@ -964,4 +964,22 @@ pub mod reply {
         pub transaction_hash: StarknetTransactionHash,
         pub contract_address: ContractAddress,
     }
+
+    /// Return type of transaction fee estimation
+    #[serde_as]
+    #[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+    #[serde(deny_unknown_fields)]
+    pub struct FeeEstimate {
+        /// The Ethereum gas cost of the transaction
+        #[serde_as(as = "crate::rpc::serde::H256AsHexStr")]
+        #[serde(rename = "gas_consumed")]
+        pub consumed: web3::types::H256,
+        /// The gas price (in gwei) that was used in the cost estimation (input to fee estimation)
+        #[serde_as(as = "crate::rpc::serde::H256AsHexStr")]
+        pub gas_price: web3::types::H256,
```

Comment:
Note that the python side has no such bounds for types, there the integers are arbitrary long. These are not affected by the serialization bounds you read the u128 limit from. The `eth_gasPrice` value we read for the whole thing, which works as a coefficient within cairo code is u256.

If anything, might want to unwind the gasprice limits a bit since we now know how that value is used.